lib/python/operations.cpp

```diff
@@ -93,6 +94,26 @@ void init_operations(py::module &m) {
   bind_allsorted(m);
   bind_midpoints(m);
 
+  m.def(
+      "get_slice_params",
+      [](const std::vector<std::string> &dims,
+         const std::vector<scipp::index> &shape, const Variable &coord,
+         const Variable &value) {
+        const auto [dim, index] =
+            get_slice_params(make_dims(dims, shape), coord, value);
+        return std::tuple{dim.name(), index};
+      },
+      py::call_guard<py::gil_scoped_release>());
+  m.def("get_slice_params",
+        [](const std::vector<std::string> &dims,
+           const std::vector<scipp::index> &shape, const Variable &coord,
+           const Variable &begin, const Variable &end) {
+          const auto [dim, start, stop] =
+              get_slice_params(make_dims(dims, shape), coord, begin, end);
+          // Do NOT release GIL since using py::slice
+          return std::tuple{dim.name(), py::slice(start, stop, 1)};
+        });
```

Comment:
Not sure it matters, but we could release the GIL and acquire it just before calling `py::slice`. Or return ints from C++ and make the slice in a Python wrapper.